### PR TITLE
OCPBUGS-7190: Reuse LDAP connection when performing group sync

### PR DESCRIPTION
**Pull request by stlaz**
The code was written more like a kube-client but it makes more sense to reuse the connection to the LDAP server for successive queries to prevent LDAP conn/bind spamming.

/hold
fake bumps

/assign @ibihim

### CODE REVIEW

File: pkg/cli/admin/groups/sync/ad.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	legacyconfigv1 "github.com/openshift/api/legacyconfig/v1"
 	"github.com/openshift/library-go/pkg/security/ldapclient"
 	ldapquery "github.com/openshift/library-go/pkg/security/ldapquery"
-	"github.com/openshift/oc/pkg/helpers/groupsync"
+	syncgroups "github.com/openshift/oc/pkg/helpers/groupsync"
```
Comment (ibihim) [inline pkg/cli/admin/groups/sync/ad.go:7]: How did it work before? I see that `syncgroups` is referenced in the file, but it was not defined before.

Comment (stlaz) [reply pkg/cli/admin/groups/sync/ad.go:7]: I have no idea, this was an automatic change

**Review by ibihim — APPROVED** [review]
/lgtm

**Review by ibihim — CHANGES_REQUESTED** [review]
We need to close the connection at some point...

File: pkg/cli/admin/groups/sync/ad.go (hunk 1 above)
Comment (ibihim) [reply pkg/cli/admin/groups/sync/ad.go:7]: witchcraft

**Comment by ardaguclu** [comment]
Idea behind the changes look good to me. I think we need a pre-merge testing for this after PR is ready to be merged(after applying real bump instead fake bump)

**Comment by stlaz** [comment]
Thanks! There is in fact group sync testing as a part of the e2e serial test suite. We should go ahead with the library-go changes, then :+1:

**Comment by zhouying7780** [comment]
Done pre-merge test . no issue found . see details from the bug .

**Pull request merged**